### ISSUE TITLE
"Could not resolve plugin "tiledquickplugin"" when running application that imports "org.mapeditor.Tiled 1.0" when built with qbs on macOS

### ISSUE THREAD

**Issue by mitchcurtis** [bjorn/tiled#1945]
I'm porting my game over to qbs as I've read that Tiled will move away from qmake completely at some stage.

I have everything working on macOS, except that I need to rename `tiledquickplugin` to `libtiledquickplugin.dylib` so that the QML import works. If I don't, I get this: 

```
17:381 QQmlImports::addLibraryImport: QQmlImports(qrc:/qml/GameScreen.qml)::addLibraryImport: "org.mapeditor.Tiled" 1.0 as ""
17:381 QQmlImportsPrivate::importExtension: QQmlImports(qrc:/qml/GameScreen.qml)::importExtension: loaded "/Users/mitch/dev/tshnm-iso-qt5_isle_fw-Debug/qtc_qt5_isle_fw_Debug/install-root/qml/org/mapeditor/Tiled/qmldir"
17:381 QQmlImportDatabase::resolvePlugin: QQmlImportDatabase::resolvePlugin: Could not resolve plugin "tiledquickplugin" in "/Users/mitch/dev/tshnm-iso-qt5_isle_fw-Debug/qtc_qt5_isle_fw_Debug/install-root/qml/org/mapeditor/Tiled"
17:381 QQmlComponent::create: QQmlComponent: Component is not ready
```

Qt searches for `_debug.dylib`, `.dylib`, `.bundle`, and `.so` extensions when looking for the plugin.

I checked other Qt plugins on my machine and they all follow the `libqtfooplugin.dylib` format.
 
What's weird is that the [targetName](https://github.com/bjorn/tiled/blob/wip/qtquick/src/tiledquickplugin/tiledquickplugin.qbs#L4) is specified the same way as e.g. [libtiled](https://github.com/bjorn/tiled/blob/master/src/libtiled/libtiled.qbs#L4) (which is correctly named `libtiled.dylib`), and they're both `DynamicLibrary`, so I don't know what makes `tiledquickplugin` come out differently.

**Comment by bjorn** [comment]
Hmm, that's very strange indeed. I'll try this on macOS next week.

**Comment by mitchcurtis** [comment]
One thing I forgot to mention is that this only happens on macOS: on Windows and Linux it works without problems (I get the expected .dll and .so extensions on those plaftorms).

**Comment by mitchcurtis** [comment]
Did you ever get around to trying this out?

**Comment by bjorn** [comment]
Sorry, I forgot. I'll try it tomorrow!

**Comment by bjorn** [comment]
So yeah, I didn't get around to it after my Mac Mini had finished restarting for updates. Another chance on Monday!

**Comment by mitchcurtis** [comment]
Haha, no worries!

**Comment by bjorn** [comment]
This turned out to be because the plugin was built as a framework. I've pushed a fix to the `wip/qtquick` branch.

However, I was personally still unable to run Tiled Quick out of the box. This appears to be due to a problem with the install locations of Tiled Quick and the QML plugin, which causes the following output when trying to do `qbs run -p tiledquick`:

```
QQmlApplicationEngine failed to load component
qrc:/main.qml:5 module "org.mapeditor.Tiled" is not installed
```

But judging by your error you had already gotten past this problem.

**Comment by bjorn** [comment]
I tried to improve the situation a little in change b3976987f1f18130a35c316351072a603a67f2c3, but it's still not actually loading this plugin out of the box, because it also depends on `libtiled.dylib`, which is installed into `Tiled.app` and I could not find a way to install it to `Tiled Quick.app` as well.

**Comment by bjorn** [comment]
Closing this issue just because that particular problem was resolved, but getting this working out-of-the-box remains to be done.

**Comment by mitchcurtis** [comment]
Thanks, I'll try it out.

**Comment by mitchcurtis** [comment]
Oh wait, I don't have https://github.com/bjorn/tiled/commit/b3976987f1f18130a35c316351072a603a67f2c3...

**Comment by mitchcurtis** [comment]
I can't get it to work either, and I'm having trouble figuring out why.

One option is to just make it a library instead, then just register the types in the application. That would actually be nice for me too, because then I could use the types in C++ (whereas I'm currently treating them as QObjects and using invokeMethod()).

**Comment by mitchcurtis** [comment]
I created https://github.com/bjorn/tiled/issues/1967 as a suggestion for that. :D

**Comment by bjorn** [comment]
So, the copying of `libtiled.dylib` for both Tiled and Tiled Quick is now solved with change 6d2741807f709501748e47ee9c4a7cc8dafc3923. This means Tiled Quick now runs out of the box on macOS! :-)
